src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -458,42 +473,53 @@ private bool HandleError(C4Error error)
             return true;
         }
 
-        private void OnDocEnded(List<ReplicatedDocument> replications, bool pushing)
+        private void OnDocEnded(ConcurrentBag<ReplicatedDocument> replications, bool pushing)
         {
             if (_disposed) {
                 return;
             }
 
-            for(int i = 0; i < replications.Count; i++) {
+            while (!replications.IsEmpty) {
+                _conflictTasks.Add(Task.Run(() =>
+                {
+                    ReplicatedDocument replication;
+                    if (replications.TryTake(out replication)) {
+                        try {
+                            Config.Database.ResolveConflict(replication.Id, Config.ConflictResolver);
+                            replication = replication.ClearError();
+                        } catch (Exception e) {
+                            C4Error error;
+                            if(e.GetBaseException() is CouchbaseException) {
+                                var errCode = ((CouchbaseException)e).Error;
+                                error = new C4Error((C4ErrorCode)errCode);
+                            } else {
+                                error = new C4Error(C4ErrorCode.UnexpectedError);
+                            }
+                            replication = replication.AssignError(error);
+                            WriteLog.To.Sync.W(Tag, $"Conflict resolution of '{replication.Id}' failed", e);
+                        }
+                    }
+                    _documentEndedUpdate.Fire(this, new DocumentReplicationEventArgs(new[] { replication }, pushing));
+                }).ContinueWith(task => _conflictTasks.TryTake(out var t)));
```

Comment:
`TryTake` does not do what you think it does.  This will remove a random element from the bag, so probably you've just skipped things by removing a different task.  This is the reason for making it a concurrent dictionary (there is no way to remove a specific item from a concurrent bag)

Reply:
Are we able to reset _conflictTasks after waitAll call?